packages/shopping/src/middlewares/shoppy.middleware.ts

```diff
@@ -0,0 +1,16 @@
+import {Middleware, MiddlewareContext, RestBindings} from '@loopback/rest';
+import {Next} from '@loopback/context';
+
+export const ShoppyMiddleware: Middleware = async (
+  ctx: MiddlewareContext,
+  next: Next,
+) => {
+  const send = await ctx.get(RestBindings.SequenceActions.SEND);
+  const reject = await ctx.get(RestBindings.SequenceActions.REJECT);
+  try {
+    const result = await next();
+    send(ctx.response, result);
+  } catch (error) {
+    reject(ctx, error);
+  }
```

Comment:
This looks identical to the [SendResponseMiddlewareProvider](https://github.com/strongloop/loopback-next/blob/1fe4c274efb32210699c13be38d8b71365bee9c4/packages/rest/src/providers/send.provider.ts#L42-L55), which is included in the [default REST Middleware Chain](https://github.com/strongloop/loopback-next/blob/1fe4c274efb32210699c13be38d8b71365bee9c4/packages/rest/src/sequence.ts#L195-L198).

Since the default Middleware Sequence is used, there shouldn't be a need for a custom middleware.

Reply:
Closing PR since I won't have time to look into this at the moment

Reply:
@achrinza the objective is to show the use of the middleware hence I was leaning on @raymondfeng providing a use case for such as well as for the provider. Most users don't understand how these fit together even from the official documentation it is not clear. Looking at how we missed this the first time I assume we are also not fully familiar with how this works

Reply:
@raymondfeng please help with a use case, which I can then use to update the documentation on this topic. Am sure from an architecture perspective there was a good reason to switch from a default action-based sequence to this new approach. I would like to make it clear to our users

Reply:
I think it might add to the confusion as readers may assume that this is a necessary step, when in reality, it's not necessary as there's already a default middleware provider handling this. The concern is that developers may blindly copy this even though it's not the "correct" usage.

In this case, improving the documentation would be the more effective solution.

Reply:
I think it might add to the confusion as readers may assume that this is a necessary step, when in reality, it's not necessary as there's already a default middleware provider handling this. The concern is that developers may blindly copy this even though it's not the "correct" usage.

In this case, improving the documentation would be the more effective solution.

Reply:
I think we missed each other. What am referring to on the above is I want to demonstrate how the Sequence, Middleware and possibly middleware-provider fit together not necessarily use the above changes

Reply:
@achrinza I found how the error handling is done in the middleware-based sequence. Instead of recreating the middleware like I did earlier one rather creates a Middleware-provider that somewhat has a middleware in it. See the updated changes. I have to admit its a lot though unlike the previous action-based. I

Also, I figured out why we were missing each other. My fault, I kind of assumed we were on the same page. Context, we had error handling implemented in the action-based sequence, as such when I was migrating to middleware-base sequence I also wanted to demonstrate how does one handle errors. Trust me this can be a daunting task I personally went through it and based on Raymond’s link above I just discovered I went about it the wrong way. The correct way is in the updated changes
 
PTAL @raymondfeng